### PR TITLE
Improve ACL performance by only adding in the OR clause if there are …

### PR DESCRIPTION
**Pull request by seamuslee001**
…any contacts stored in the table

ping @eileenmcnaughton @kcristiano

### CODE REVIEW

File: relatedpermissions.php [hunk 1]
```diff
@@ -149,15 +149,18 @@ function relatedpermissions_civicrm_aclWhereClause($type, &$tables, &$whereTable
...
-    $tables['$tmpTableName'] = $whereTables['$tmpTableName'] =
-      " LEFT JOIN $tmpTableName permrelationships
-     ON (contact_a.id = permrelationships.contact_id)";
-    if (empty($where)) {
-      $where = " permrelationships.contact_id IS NOT NULL ";
-    }
-    else {
-      $where = '(' . $where . " OR permrelationships.contact_id IS NOT NULL " . ')';
+    // Do not add in the permission table join and associated OR clause if there are no permitted relationships
+    $check = CRM_Core_DAO::singleValueQuery("SELECT count(contact_id) as contact_count FROM {$tmpTableName}");
```
Comment (eileenmcnaughton) [inline relatedpermissions.php:153]: Won't this mean they get too many permissions if no rows exist?

Comment (seamuslee001) [reply relatedpermissions.php:153]: Maybe but only if there is no other where right? like if we have a where clause we are saying check that where OR this thing and its the or that is really the hurting part

**Comment by seamuslee001** [comment]
@eileenmcnaughton I have added in a unit test to try to cover this change. is there anything that you think needs to be added to cover this off?

**Comment by eileenmcnaughton** [comment]
@seamuslee001 I'll take your word for it @aydun @MegaphoneJon you might like to test with this before the next release is tagged

**Pull request merged**